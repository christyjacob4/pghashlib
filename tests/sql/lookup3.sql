-- Test basic hash functionality with text
SELECT lookup3('hello world');

-- Test hash with different text inputs
SELECT lookup3('test string');
SELECT lookup3('another test');

-- Test text input with custom initval
SELECT lookup3('hello world', 42);
SELECT lookup3('hello world', 123);

-- Test bytea input
SELECT lookup3('hello world'::bytea);

-- Test bytea input with custom initval
SELECT lookup3('hello world'::bytea, 42);

-- Test integer input
SELECT lookup3(12345);
SELECT lookup3(-12345);

-- Test integer input with custom initval
SELECT lookup3(12345, 42);
SELECT lookup3(-12345, 123);

-- Test consistency (same input should give same hash)
SELECT lookup3('consistent test') = lookup3('consistent test');

-- Test initval effect (same input, different initvals should give different hashes)
SELECT lookup3('initval test', 1) != lookup3('initval test', 2);

-- Test function properties
SELECT 
    proname,
    provolatile,
    proisstrict
FROM pg_proc 
WHERE proname = 'lookup3'
ORDER BY proname, proargtypes;

-- Test extension metadata
SELECT 
    extname,
    extversion
FROM pg_extension 
WHERE extname = 'hashlib';