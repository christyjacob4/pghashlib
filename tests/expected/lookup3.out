-- Test basic hash functionality with text
SELECT lookup3('hello world');
  lookup3   
------------
 1252609637
(1 row)

-- Test hash with different text inputs
SELECT lookup3('test string');
  lookup3  
-----------
 546331042
(1 row)

SELECT lookup3('another test');
   lookup3   
-------------
 -2135089103
(1 row)

-- Test text input with custom initval
SELECT lookup3('hello world', 42);
  lookup3   
------------
 -419950101
(1 row)

SELECT lookup3('hello world', 123);
  lookup3   
------------
 -656773485
(1 row)

-- Test bytea input
SELECT lookup3('hello world'::bytea);
  lookup3   
------------
 1252609637
(1 row)

-- Test bytea input with custom initval
SELECT lookup3('hello world'::bytea, 42);
  lookup3   
------------
 -419950101
(1 row)

-- Test integer input
SELECT lookup3(12345);
  lookup3   
------------
 1490861095
(1 row)

SELECT lookup3(-12345);
  lookup3   
------------
 -983645695
(1 row)

-- Test integer input with custom initval
SELECT lookup3(12345, 42);
  lookup3   
------------
 -483149855
(1 row)

SELECT lookup3(-12345, 123);
   lookup3   
-------------
 -1537810365
(1 row)

-- Test consistency (same input should give same hash)
SELECT lookup3('consistent test') = lookup3('consistent test');
 ?column? 
----------
 t
(1 row)

-- Test initval effect (same input, different initvals should give different hashes)
SELECT lookup3('initval test', 1) != lookup3('initval test', 2);
 ?column? 
----------
 t
(1 row)

-- Test function properties
SELECT 
    proname,
    provolatile,
    proisstrict
FROM pg_proc 
WHERE proname = 'lookup3'
ORDER BY proname, proargtypes;
 proname | provolatile | proisstrict 
---------+-------------+-------------
 lookup3 | i           | t
 lookup3 | i           | t
 lookup3 | i           | t
 lookup3 | i           | t
 lookup3 | i           | t
 lookup3 | i           | t
(6 rows)

-- Test extension metadata
SELECT 
    extname,
    extversion
FROM pg_extension 
WHERE extname = 'hashlib';
 extname | extversion 
---------+------------
 hashlib | 0.0.1
(1 row)
